starcounter-include.html

```diff
@@ -32,13 +32,74 @@
             static get observedAttributes() {
                 return ['partial', 'view-model', 'partialId'];
             }
-            // the self argument might be provided or not
-            // in both cases, the mandatory `super()` call
-            // will return the right context/instance to use
-            // and eventually return
-            constructor(self) {
-                self = super(self);
-                this.createComponent();
+            /**
+            * Create shadowRoot, define property setters, set initial partial.
+            */
+            constructor() {
+                super();
+                this.defaultComposition = null;
+                var partialId = this.partialId;
+                var partial = this.viewModel || this.partial || undefined;
+                checkForNonNamespaced(partial, this);
```

Comment:
Is this needed? It is repeated on line 61.

Reply:
It is. Line 61 is in setter. And partial property could be set before setter is defined.

Reply:
Can we do this somehow differently, e.g. `this.partial = this.partial` at the end of the constructor?

Reply:
Why? isn't it the more explicit way it is?

Reply:
`this.partial = this.partial` is the first line I'd remove thinking it does nothing.

Reply:
For me this looks un-DRY. But let's keep it like that, I don't want to block the PR any further.